### ISSUE TITLE
ERROR] Invalid option in build() call: "incremental"

### ISSUE THREAD

**Issue by ianzone** [floydspace/serverless-esbuild#427]
**Describe the bug**
```
> nest build && sls offline

✘ [ERROR] Invalid option in build() call: "incremental"

    /home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:255:12:
      255 │       throw new Error(`Invalid option ${where}: ${quote(key)}`);
          ╵             ^

    at checkForInvalidFlags (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:255:13)
    at flagsForBuildOptions (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:457:3)
    at buildOrContextContinue (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:995:9)
    at buildOrContextImpl (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:980:5)
    at Object.buildOrContext (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:763:5)
    at /home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:2122:15
    at new Promise (<anonymous>)
    at Object.build (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:2121:25)
    at build (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:1970:51)
    at bundleMapper (/home/ian/templates/lambda-nest/node_modules/.pnpm/serverless-esbuild@1.37.0_esbuild@0.17.0/node_modules/serverless-esbuild/src/bundle.ts:98:31)
    at /home/ian/templates/lambda-nest/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:28

✘ [ERROR] Invalid option in build() call: "incremental"

    /home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:255:12:
      255 │       throw new Error(`Invalid option ${where}: ${quote(key)}`);
          ╵             ^

    at checkForInvalidFlags (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:255:13)
    at flagsForBuildOptions (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:457:3)
    at buildOrContextContinue (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:995:9)
    at buildOrContextImpl (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:980:5)
    at Object.buildOrContext (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:763:5)
    at /home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:2122:15
    at new Promise (<anonymous>)
    at Object.build (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:2121:25)
    at build (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:1970:51)
    at bundleMapper (/home/ian/templates/lambda-nest/node_modules/.pnpm/serverless-esbuild@1.37.0_esbuild@0.17.0/node_modules/serverless-esbuild/src/bundle.ts:98:31)
    at /home/ian/templates/lambda-nest/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:28

Environment: linux, node 18.12.0, framework 3.26.0, plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Error: Build failed with 1 error:
/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:255:12: ERROR: Invalid option in build() call: "incremental"
    at failureErrorWithLog (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:1596:15)
    at /home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:940:16
    at responseCallbacks.<computed> (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:684:9)
    at handleIncomingPacket (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:739:9)
    at Socket.readFromStdout (/home/ian/templates/lambda-nest/node_modules/.pnpm/esbuild@0.17.0/node_modules/esbuild/lib/main.js:660:7)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Socket.Readable.push (node:internal/streams/readable:234:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)
 ELIFECYCLE  Command failed with exit code 1.
```

**To Reproduce**
https://github.com/ianzone/serverless-nestjs-template/tree/esbuild-failed

1. Run `degit ianzone/serverless-nestjs-template#esbuild-failed`
2. Run `pnpm i`
3. Run `pnpm run start`

**Expected behavior**
app runs

**Versions (please complete the following information):**

- OS: Ubuntu 22.04.1 LTS on Windows 10 x86_64
- Serverless Framework Version: 3.26.0
- Plugin Version: 1.37.0
- Esbuild Version: 0.17.0

**Comment by efstathiosntonas** [comment]
same here, out of the sudden all my github and `serverless-offline` started failing.

**Comment by efstathiosntonas** [comment]
downgrading `esbuild@0.16.17` to `0.16.16` solved the issue for me

**Comment by JacksonGariety** [comment]
I'm also seeing this.

**Comment by KillDozerX2** [comment]
I'm also seeing this and using `disableIncremental` doesn't help either. Downgrading to 0.16.16 works.

**Comment by timoaudi** [comment]
Still seeing this issue in `0.17.3`

**Comment by pencilcheck** [comment]
Still seeing this in 0.17.x as well

**Comment by AhmedNadar** [comment]
Had the same issue with `"esbuild": "^0.17.3"` and after downgrading to `"esbuild": "^0.16.17"` my app worked fine.

**Comment by efstathiosntonas** [comment]
@AhmedNadar the `^` will install 0.17, either use ~ or @0.16.17 to pin the version.

**Comment by FilipPyrek** [comment]
There has been a breaking change in esbuild 0.17.0.

More on that here: https://github.com/evanw/esbuild/blob/main/CHANGELOG.md#0170

> **This release deliberately contains backwards-incompatible changes.** To avoid automatically picking up releases like this, you should either be pinning the exact version of `esbuild` in your `package.json` file (recommended) or be using a version range syntax that only accepts patch upgrades such as `^0.16.0` or `~0.16.0`. See npm's documentation about [semver](https://docs.npmjs.com/cli/v6/using-npm/semver/) for more information.
>
>At a high level, the breaking changes in this release fix some long-standing issues with the design of esbuild's **incremental**, watch, and serve APIs. This release also introduces some exciting new features such as live reloading. In detail:

**Comment by FilipPyrek** [comment]
@jlangsu's PR looks promising 🤞

**Comment by jlangsu** [comment]
If the build fails on an incremental build, 0.17.x+ #432 doesn't dispose of the context just yet, causing the serverless process to stay running.

I'm working to push another commit to account for this.

**Comment by jlangsu** [comment]
#432 merged in release v1.38.0.

**Comment by efstathiosntonas** [comment]
Iis anyone else getting `✖ Bundle error, waiting for a file change to reload...` while using 1.38.0? 

`tsc --noEmit` does not throw any errors.
